quinn-proto/src/endpoint.rs

```diff
@@ -772,13 +778,10 @@ impl Endpoint {
     /// This does not ensure delivery of outstanding data. It is the application's responsibility
     /// to call this only when all important communications have been completed.
     pub fn close(&mut self, now: Instant, ch: ConnectionHandle, error_code: u16, reason: Bytes) {
-        if self.connections[ch].conn.is_drained() {
-            self.forget(ch);
-            return;
-        }
         self.connections[ch].conn.close(now, error_code, reason);
         self.connections[ch].app_closed = true;
         self.needs_transmit.insert(ch);
+        self.endpoint_eventful_conns.insert(ch);
```

Comment:
Let's try to keep documentation for times when a connection might have new endpoint events, in the same vein as for the existing `poll_*` functions.

Reply:
The reason I did not spend much time on it is that all the connection-tracking bits in `Endpoint` (so `dirty_timers`, `needs_transmit`, `endpoint_eventful_conns`, `eventful_conns`) are going away pretty soon, so spending much more attention on this didn't seem worth it.

Reply:
Fair enough!